samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -0,0 +1,164 @@
+	}
+
+	return PersistentStorage::Instance().Load(&mBridgedDevicesIndexes, indexes, maxCount, count);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceEndpointId(uint16_t endpointId, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return PersistentStorage::Instance().Store(&id, &endpointId, sizeof(endpointId));
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceEndpointId(uint16_t &endpointId, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return LoadSimpleData(&id, endpointId);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceEndpointId(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceEndpointId);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceNodeLabel(const char *label, size_t labelLength,
+						       uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Store(&id, label, labelLength);
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceNodeLabel(char *label, size_t labelMaxLength, size_t &labelLength,
+						      uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Load(&id, label, labelMaxLength, labelLength);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceNodeLabel(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceNodeLabel);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDeviceType(uint16_t deviceType, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return PersistentStorage::Instance().Store(&id, &deviceType, sizeof(deviceType));
+}
+
+bool BridgeStorageManager::LoadBridgedDeviceType(uint16_t &deviceType, uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return LoadSimpleData(&id, deviceType);
+}
+
+bool BridgeStorageManager::RemoveBridgedDeviceType(uint8_t bridgedDeviceIndex)
+{
+	PersistentStorageNode id = CreateIndexNode(bridgedDeviceIndex, &mBridgedDeviceType);
+
+	return PersistentStorage::Instance().Remove(&id);
+}
+
+bool BridgeStorageManager::StoreBridgedDevice(BridgedDevice *device, uint8_t index)
+{
+	if (!device) {
+		return false;
+	}
+
+	if (!StoreBridgedDeviceEndpointId(static_cast<uint16_t>(device->GetEndpointId()), index)) {
+		return false;
+	}
+
+	if (!StoreBridgedDeviceNodeLabel(device->GetNodeLabel(), strlen(device->GetNodeLabel()), index)) {
+		return false;
+	}
+
+	return StoreBridgedDeviceType(static_cast<uint16_t>(device->GetDeviceType()), index);
```

Comment:
Do we need the cast here? the `DeviceType` is a strongly uint16_t-typed enum (not a class enum), so I'd expect the cast should be implicit here, right?